gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -1030,6 +1031,22 @@ public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
         }
       };
 
+  public static final TypeAdapterFactory RAW_ENUM_FACTORY =
+      new TypeAdapterFactory() {
+        @Override
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+          Class<? super T> rawType = typeToken.getRawType();
+          if (rawType == Enum.class) {
+            @SuppressWarnings("unchecked")
+            TypeAdapter<T> adapter =
+                (TypeAdapter<T>) new ObjectTypeAdapter(gson, ToNumberPolicy.DOUBLE);
+            return adapter;
+          } else {
+            return null;
+          }
+        }
+      };
```

Comment:
While this might be functionally equivalent to the previous behavior, I think if we are going to add an adapter for `java.lang.Enum` anyway, then we should implement it properly and have it behave explicitly behave as desired (instead of implicitly relying on `ObjectTypeAdapter`'s behavior.

Something like this should work:
```java
public static final TypeAdapterFactory ENUM_BASE_CLASS_FACTORY = new TypeAdapterFactory() {
  @Override
  public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
    if (type.getRawType() != Enum.class) {
      return null;
    }

    return new TypeAdapter<T>() {
      @Override
      public T read(JsonReader in) throws IOException {
        throw new UnsupportedOperationException("Deserializing base class java.lang.Enum is not supported;"
            + " only deserialization of subclasses is supported");
      }

      @Override
      public void write(JsonWriter out, T value) throws IOException {
        if (value == null) {
          out.nullValue();
          return;
        }

        @SuppressWarnings("unchecked")
        TypeAdapter<T> delegate = (TypeAdapter<T>) gson.getAdapter(value.getClass());
        delegate.write(out, value);
      }
    };
  }
};
```